torchao/quantization/quant_api.py

```diff
@@ -468,7 +480,10 @@ def insert_subclass(lin):
 
 def quantize_(
     model: torch.nn.Module,
-    apply_tensor_subclass: Callable[[torch.nn.Module], torch.nn.Module],
+    # apply_tensor_subclass: Callable[[torch.nn.Module], torch.nn.Module],
+    apply_tensor_subclass: Union[
+        Callable[[torch.nn.Module], torch.nn.Module], AOBaseWorkflowConfig
```

Comment:
I think there are two BC breaking steps:
1. Change the arg name to "config"
2. Change the arg type to `AOBaseWorkflowConfig` only

Based on my quick search I feel we can just do (1) now. I haven't found any use cases yet that call this with a keyword argument, so this seems relatively safe. For (2) maybe we should deprecate with warning first and remove after a couple releases? I feel this use case (users passing in arbitrary functions) is harder to grep for.

Curious what others think as well @drisspg @HDCharles